ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java

```diff
@@ -158,15 +188,10 @@ private HttpClientRequest configureBasicAuth(final HttpClientRequest request) {
   private static <T> void handleResponse(
       final HttpClientResponse response,
       final CompletableFuture<T> cf,
```

Comment:
Seems odd to pass in both a cf and a Handler<HttpClientResponse> to signal back both exceptionally and successful results. Can't the cf do both?

Reply:
Refactored this to be less convoluted.